### PR TITLE
Update handling of flashbots flow

### PR DESCRIPTION
**Pull request by jinchung**
Fixes APP-2036
Fixes APP-2037

## What changed (plus any additional context for devs)
- The swap and crosschain swap actions do one final gas estimation before submitting the transaction. The function they were using was not taking into account the metadata endpoint which has more accurate gas predictions. Moved them both over to use this function instead.
- The fallback logic of trying different gas limits for estimating gas was failing because trace call complains about formatting of block tag or block number
- The `getNonceAndPerformSwap` function was not actually getting the nonce, causing the provider to default to its own calculated nonce, which was causing issues once we removed the "wait for node ack"

### CODE REVIEW

File: src/handlers/swap.ts [hunk 1]
```diff
@@ -184,73 +176,6 @@ export const getSwapGasLimitWithFakeApproval = async (
   return getDefaultGasLimitForTrade(tradeDetails, chainId);
 };
 
-export const estimateSwapGasLimit = async ({
```
Comment (jinchung) [inline src/handlers/swap.ts:187]: unused from swaps v1 - removed

File: src/raps/actions/crosschainSwap.ts [hunk 2]
```diff
@@ -27,6 +30,65 @@ import { chainsName } from '@/chains';
 
 const getCrosschainSwapDefaultGasLimit = (quote: CrosschainQuote) => quote?.routes?.[0]?.userTxs?.[0]?.gasFees?.gasLimit;
 
+export const estimateUnlockAndCrosschainSwap = async ({
```
Comment (jinchung) [inline src/raps/actions/crosschainSwap.ts:33]: this function was moved over from `unlockAndCrosschainSwap`

File: src/raps/actions/crosschainSwap.ts [hunk 3]
```diff
@@ -125,10 +187,11 @@ export const crosschainSwap = async ({
 
   let gasLimit;
   try {
-    gasLimit = await estimateCrosschainSwapGasLimit({
-      chainId,
-      requiresApprove,
+    gasLimit = await estimateUnlockAndCrosschainSwap({
```
Comment (jinchung) [inline src/raps/actions/crosschainSwap.ts:190]: main change is to use this function instead, which takes into account the metadata estimations

File: src/raps/actions/swap.ts [hunk 4]
```diff
@@ -45,6 +45,73 @@ import { chainsName } from '@/chains';
 
 const WRAP_GAS_PADDING = 1.002;
 
+export const estimateUnlockAndSwap = async ({
```
Comment (jinchung) [inline src/raps/actions/swap.ts:48]: this function was moved over from `unlockAndSwap`

File: src/raps/actions/swap.ts [hunk 5]
```diff
@@ -251,9 +318,10 @@ export const swap = async ({
 
   let gasLimit;
   try {
-    gasLimit = await estimateSwapGasLimit({
+    gasLimit = await estimateUnlockAndSwap({
```
Comment (jinchung) [inline src/raps/actions/swap.ts:321]: main change is to use this function instead, which takes into account the metadata estimations

**Comment by brunobar79** [comment]
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@jin/gas-blocktag/d0f2cf5a0b147b8360a90c4b8cc35a9d41457441.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@jin/gas-blocktag/d0f2cf5a0b147b8360a90c4b8cc35a9d41457441.ipa) for d0f2cf5a0b147b8360a90c4b8cc35a9d41457441

**Comment by brunobar79** [comment]
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@jin/gas-blocktag/8a6378539723c586f33367353226e08bddcd9b39.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@jin/gas-blocktag/8a6378539723c586f33367353226e08bddcd9b39.ipa) for 8a6378539723c586f33367353226e08bddcd9b39

**Pull request merged**